File: src/main/java/nl/guno/intellij/savechangelisttopatches/SaveChangeListsToPatchesApplicationComponent.java
package nl.guno.intellij.savechangelisttopatches;

import org.jetbrains.annotations.NotNull;

import com.intellij.openapi.components.ProjectComponent;
import com.intellij.openapi.project.Project;
import nl.guno.intellij.savechangelisttopatches.settings.Settings;

public class SaveChangeListsToPatchesApplicationComponent implements ProjectComponent {

    private Project project;

    public SaveChangeListsToPatchesApplicationComponent(Project project) {
        this.project = project;
    }

    @Override
    public void projectOpened() {}

    @Override
    public void projectClosed() {
        if (Settings.getInstance(project).getSaveOnClose()) {
            new ChangeListsSaver(project).savePatches();
        }
    }

    @Override
    public void initComponent() {}

    @Override
    public void disposeComponent() {}

    @Override
    @NotNull
    public String getComponentName() {
        return "SaveChangeListsToPatchesApplicationComponent";
    }


}

File: src/main/kotlin/nl/guno/intellij/savechangelisttopatches/SaveChangeListsToPatchesApplicationComponent.kt
package nl.guno.intellij.savechangelisttopatches

import com.intellij.openapi.components.ProjectComponent
import com.intellij.openapi.project.Project
import nl.guno.intellij.savechangelisttopatches.settings.Settings

class SaveChangeListsToPatchesApplicationComponent(private val project: Project) : ProjectComponent {

    override fun projectOpened() {}

    override fun projectClosed() {
        if (Settings.getInstance(project).saveOnClose) {
            ChangeListsSaver(project).savePatches(true)
        }
    }

    override fun initComponent() {}

    override fun disposeComponent() {}

    override fun getComponentName(): String {
        return "SaveChangeListsToPatchesApplicationComponent"
    }


}
